tests/polars/test_polars_components.py

```diff
@@ -194,6 +196,47 @@ def test_check_dtype(data, from_dtype, check_dtype):
         )
 
 
+def test_check_data_container():
+    @polars_engine.Engine.register_dtype
+    class MyTestStartsWithID(polars_engine.String):
+        """
+        Test DataType which expects strings starting with "id_"
+        """
+
+        def check(
+            self,
+            pandera_dtype: DataType,
+            data_container: Optional[polars_engine.PolarsDataContainer] = None,
```

Comment:
the type needs to be `PolarsData` no?

Reply:
Ah, was following how I saw other dtypes take the parameter in. https://github.com/unionai-oss/pandera/blob/612d25c6541d426355195e0e3eb613dbc1a6c67d/pandera/engines/polars_engine.py#L370-L371
It looks like when it needs to be treated as PolarsData, it may be wrapped in the class. Think I'm going to handle mypy error similar to 
https://github.com/unionai-oss/pandera/blob/612d25c6541d426355195e0e3eb613dbc1a6c67d/pandera/engines/polars_engine.py#L632-L635

Reply:
Decided not to take this approach due to it just being for a specific test, updated to PolarsData